ath79: glinet_gl-ar150: add I2C GPIO support in DTS

olijf opened the pull request:
This patch adds an I2C bus based on the i2c-gpio driver to the GL.iNet GL-AR150 device tree. The device uses GPIO pins 1 and 17 configured as SDA and SCL to support external I2C peripherals such as the DS1307 RTC.

The configuration has been tested on real hardware with a DS1307 RTC module connected to GPIOs 1/17. The RTC is correctly detected and `hwclock` works as expected.

raenye commented:
This is not to support some built-in hardware, right?
Probably better to have this as a DT overlay, using [this](https://github.com/openwrt/packages/pull/26832).

olijf committed:
ath79: glinet_gl-ar150: add I2C GPIO support in DTS

This patch adds an I2C bus based on the i2c-gpio driver to the
GL.iNet GL-AR150 device tree. The device uses GPIO pins 1 and 17
configured as SDA and SCL to support external I2C peripherals such
as the DS1307 RTC.

The configuration has been tested on real hardware with a DS1307
RTC module connected to GPIOs 1/17. The RTC is correctly detected
and `hwclock` works as expected.

Signed-off-by: Olaf van der Kruk <olafvdkruk@gmail.com>

raenye commented:
> dynamically load the overlay 

This is what the PR I linked to implements, for all targets.
